tests/testmacro.nim

```diff
@@ -403,9 +404,10 @@ suite "Exceptions tracking":
     expect(IOError): waitFor test3()
     waitFor test6()
 
-  test "Cannot raise invalid exception":
-    checkNotCompiles:
-      proc test3 {.async: (raises: [IOError]).} = raise newException(ValueError, "hey")
+  when not chronosHandleException:
```

Comment:
why does this test need to be disabled? the global handleException default should not affect raises-annotated functions..

Reply:
Oh... apparently it does, as this snippet starts to compile when I enable it; i.e., it seems to relax the raises constraint in raises-annotated procs as well. I thought this was intentional.

Reply:
OK, running a simple test with the raises-annotated proc above and looking at the expansions I can see that the global flag does indeed modify the exception handlers (adds handlers for `Exception` and `CatchableError`, removes handler for `IOError`).

Reply:
Hm... if I understand the implementation right:

```nim
proc decodeParams(params: NimNode): AsyncParams =
  ...
  var
    raw = false
    raises: NimNode = nil
    handleException = chronosHandleException
  ...
  if param[0].eqIdent("raises"):
    ... # doesn't reset handleException to false
  elif param[0].eqIdent("handleException"):
      handleException = param[1].eqIdent("true")
```
this will indeed simply start from the global value and then override it if `handleException` is set, but won't set it to false if there's a raises clause.

Reply:
I'm experimenting with fixing this and seeing how it impacts our code.

Reply:
OK so I've added an attempt at fixing this to this branch as it sort of fits the overall "compilation issues in v3 compatibility mode" theme, and I need those fixes to compile with the flag on anyways. :-)

Reply:
OK here goes my attempt at fixing this. I've added it to this branch as it sort of fits the overall "compilation issues in v3 compatibility mode" theme, and I'd need the previous changes to be able to compile Chronos with the flag on anyways.